tests/model/test_model.py

```diff
@@ -1916,6 +1918,23 @@ def test__handle_subscription_event_subscribers_one_user_multiple_streams(
             new_subscribers = model.stream_dict[stream_id]['subscribers']
             assert new_subscribers == expected_subscribers
 
+    @pytest.mark.parametrize('event, twenty_four_hr_format', [
+        ({'setting_name': 'twenty_four_hour_time',
+          'setting': True}, False),
+    ], ids=['twenty_four_hour_time'])
+    def test__handle_update_display_settings(self, mocker, model,
+                                             twenty_four_hr_format,
+                                             event):
+        event['type'] = 'update_display_settings'
+
+        # Test for change in time format
+        if event['setting_name'] == 'twenty_four_hour_time':
```

Comment:
This is a test for only the 24-hour update, and it's likely best to keep it at that - name it that, and if we add other settings event updates then we can write separate tests as we need them, or adapt this one. Those extra parameters are just one set, which you could put as a dict in the body of the test.

You're only testing False -> True? A lot of those parameters are constants once you focus on just this part of the event, and you can focus on the values of the settings.

Reply:
> You're only testing False -> True? A lot of those parameters are constants once you focus on just this part of the event, and you can focus on the values of the settings.

Should I add tests for checking whether `create_msg_box_list` is being called and a separate test for `formatted_local_time`?

Reply:
Both in the same test seems OK for now. I'll look again in the next iteration.